src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -52,6 +53,96 @@
+            heads.
+        image_hidden_states (`tuple(torch.FloatTensor)`, *optional*):
+            Tuple of `torch.FloatTensor` (one for the output of the image embeddings, `(batch_size, num_images,
+            sequence_length, hidden_size)`.
+
+            image_hidden_states of the model produced by the vision encoder, and optionally by the perceiver
+    """
+
+    last_hidden_state: torch.FloatTensor = None
+    past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+    image_hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class IdeficsCausalLMOutputWithPast(ModelOutput):
+    """
+    Base class for causal language model (or autoregressive) outputs.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*, returned when `labels` is provided):
+            Language modeling loss (for next-token prediction).
+        logits (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`):
+            Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+        past_key_values (:
+            obj:*List[torch.FloatTensor]*, *optional*, returned when `use_cache=True` is passed or when
+            `config.use_cache=True`): List of `torch.FloatTensor` of length `config.n_layers`, with each tensor of
+            shape `(2, batch_size, num_heads, sequence_length, embed_size_per_head)`).
+
+            Contains pre-computed hidden-states (key and values in the attention blocks) that can be used (see
+            `past_key_values` input) to speed up sequential decoding.
+        hidden_states (:
+            obj:*tuple(torch.FloatTensor)*, *optional*, returned when `output_hidden_states=True` is passed or when
+            `config.output_hidden_states=True`): Tuple of `torch.FloatTensor` (one for the output of the embeddings +
+            one for the output of each layer) of shape `(batch_size, sequence_length, hidden_size)`.
+
+            Hidden-states of the model at the output of each layer plus the initial embedding outputs.
+        attentions (:
+            obj:*tuple(torch.FloatTensor)*, *optional*, returned when `output_attentions=True` is passed or when
+            `config.output_attentions=True`): Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size,
+            num_heads, sequence_length, sequence_length)`.
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+        image_hidden_states (`tuple(torch.FloatTensor)`, *optional*):
+            Tuple of *torch.FloatTensor* (one for the output of the image embeddings, *(batch_size, num_images,
+            sequence_length, hidden_size)*.
+
```

Comment:
```suggestion
            Tuple of `torch.FloatTensor` (one for the output of the image embeddings, `(batch_size, num_images,
            sequence_length, hidden_size)`.

```
should be updated everywhere you use `*` instead of `